client/blocks/login/login-header.tsx

```diff
@@ -0,0 +1,597 @@
+import { useTranslate } from 'i18n-calypso';
+import { capitalize } from 'lodash';
+import A4APlusWpComLogo from 'calypso/a8c-for-agencies/components/a4a-plus-wpcom-logo';
+import VisitSite from 'calypso/blocks/visit-site';
+import GravatarLoginLogo from 'calypso/components/gravatar-login-logo';
+import JetpackPlusWpComLogo from 'calypso/components/jetpack-plus-wpcom-logo';
+import WooCommerceConnectCartHeader from 'calypso/components/woocommerce-connect-cart-header';
+import WPCloudLogo from 'calypso/components/wp-cloud-logo';
+import { getPluginTitle } from 'calypso/lib/login';
+import {
+	isCrowdsignalOAuth2Client,
+	isJetpackCloudOAuth2Client,
+	isA4AOAuth2Client,
+	isBlazeProOAuth2Client,
+	isGravatarFlowOAuth2Client,
+	isGravatarOAuth2Client,
+	isPartnerPortalOAuth2Client,
+} from 'calypso/lib/oauth2-clients';
+
+interface LoginHeaderProps {
+	action: string;
+	currentQuery: Record< string, string >;
+	fromSite: string | null;
+	isFromAkismet: boolean;
+	isFromMigrationPlugin: boolean;
+	isFromAutomatticForAgenciesPlugin: boolean;
+	isGravPoweredClient: boolean;
+	isGravPoweredLoginPage: boolean;
+	isJetpack: boolean;
+	isManualRenewalImmediateLoginAttempt: boolean;
+	isSignupExistingAccount: boolean;
+	isSocialFirst: boolean;
+	isWhiteLogin: boolean;
+	isWCCOM: boolean;
+	isBlazePro: boolean;
+	linkingSocialService: string;
+	oauth2Client: {
+		title: string;
+		icon: string;
+	} | null;
+	privateSite: boolean;
+	socialConnect: boolean;
+	twoStepNonce: string | null;
+	wccomFrom: string;
+	isWooJPC: boolean;
+	twoFactorAuthType: string;
+	twoFactorEnabled: boolean;
+	initialQuery: { 'client-id': string; redirect_to: string } | null;
+	getSignupLinkComponent: () => JSX.Element;
+	showContinueAsUser: () => boolean;
+}
+
+export function GetHeaderText( {
+	isSocialFirst,
+	twoFactorAuthType,
+	isManualRenewalImmediateLoginAttempt,
+	twoStepNonce = null,
+	socialConnect,
+	linkingSocialService,
+	action,
+	privateSite,
+	oauth2Client,
+	isWooJPC,
+	isFromMigrationPlugin,
+	isJetpack,
+	isWCCOM,
+	isFromAkismet,
+	isFromAutomatticForAgenciesPlugin,
+	isGravPoweredClient,
+	wccomFrom,
+	twoFactorEnabled,
+	currentQuery,
+	showContinueAsUser,
+}: {
+	isSocialFirst: boolean;
+	twoFactorAuthType: string | null;
+	isManualRenewalImmediateLoginAttempt: boolean;
+	twoStepNonce: string | null;
+	socialConnect: boolean;
+	linkingSocialService: string;
+	action: string;
+	privateSite: boolean;
+	oauth2Client: { title: string; icon: string } | null;
+	isWooJPC: boolean;
+	isFromMigrationPlugin: boolean;
+	isJetpack: boolean;
+	isWCCOM: boolean;
+	isFromAkismet: boolean;
+	isFromAutomatticForAgenciesPlugin: boolean;
+	isGravPoweredClient: boolean;
+	wccomFrom: string;
+	twoFactorEnabled: boolean;
+	currentQuery: Record< string, string >;
+	showContinueAsUser: () => boolean;
+} ): JSX.Element {
+	const translate = useTranslate();
+
+	let headerText = <></>;
+
+	if ( isSocialFirst ) {
+		headerText = <>{ translate( 'Log in to WordPress.com' ) }</>;
```

Comment:
Could it be possible to early return the translation result instead of assigning to `headerText`. It'd be a minor optimization, but it'd mean not having to check every case.

Reply:
I chose to leave the conditional logic as it was because some cases override others, and I didn't want to accidentally break a login screen that I don't even know about (I think I've got most of them by now, but keep finding new instances so can't be 100% sure 😅)